RELEASE/scripts/autoscend/iotms/mr2023.ash

```diff
@@ -868,6 +878,320 @@ void auto_handleJillOfAllTrades()
+	{
+		auto_log_info("No need to remove Citizen of a Zone");
+		return false;
+	}
+	if(have_effect($effect[Citizen of a Zone]) > 0 && !contains_text(activeCitZoneMod, goal) && item_amount($item[Soft Green Echo Eyedrop Antidote]) == 0)
+	{
+		auto_log_info("Can't remove Citizen of a Zone");
+		return false;
+	}
+	if(!contains_text(activeCitZoneMod, goal) && item_amount($item[Soft Green Echo Eyedrop Antidote]) > 0) //try to remove Citizen of a Zone
+	{
+		uneffect($effect[Citizen of a Zone]);
+		if(have_effect($effect[Citizen of a Zone]) > 0)
+		{
+			auto_log_debug("Tried to remove Citizen of a Zone but couldn't");
+			return false;
+		}
+	}
+	return true;
+}
+
+boolean auto_getCitizenZone(location loc)
+{
+	familiar eagle = $familiar[Patriotic Eagle];
+	//zones are approximately organized by autoscend level quest structure
+	boolean[location] meatZones = $locations[The Battlefield (Frat Uniform), The Hidden Hospital, The Haunted Bathroom, The Castle in the Clouds in the Sky (Basement),
+	Lair of the Ninja Snowmen, The Defiled Cranny, The Laugh Floor, The Batrat and Ratbat Burrow, The Sleazy Back Alley];
+	boolean[location] itemZones = $locations[A Massive Ziggurat, The Haunted Laundry Room, Whitey's Grove, The Icy Peak, Itznotyerzitz Mine,
+	The Dark Heart of the Woods, The Hidden Temple, The Haunted Library, The Bat Hole Entrance, Noob Cave];
+	boolean[location] initZones = $locations[The Feeding Chamber, An Unusually Quiet Barroom Brawl, An Overgrown Shrine (Northeast),
+	Oil Peak, Cobb's Knob Kitchens, The VERY Unquiet Garves, The Haunted Kitchen];
+	//mp zones are organized by 20-30 mp regen then 10-15 mp regen and then approximately autoscend level quest structure
+	boolean[location] mpZones = $locations[Sonofa Beach, The Themthar Hills, The Upper Chamber, Inside the Palindome, An Overgrown Shrine (Northwest), A-boo Peak, Hippy Camp,
+	Megalo-City, Shadow Rift, Vanya's Castle, The Hatching Chamber, Wartime Hippy Camp (Frat Disguise), Frat House, The Middle Chamber, The Black Forest,
+	The Haunted Ballroom, The Red Zeppelin, An Overgrown Shrine (Southwest), The Hidden Park, Twin Peak, The Smut Orc Logging Camp, The Daily Dungeon, The Spooky Forest];
+	boolean[location] specZones = $locations[The Outskirts of Cobb\'s Knob];
+	string activeCitZoneMod = activeCitZoneMod();
+	string goal;
+	
+	if(!can_adventure(loc))
+	{
+		return false;
+	}
+	//set goal for tracking
+	if(specZones contains loc)
+	{
+		if(auto_haveSeptEmberCenser() && turns_played() == 0) //ideally also have spring away
+		{
+			goal = "spec";
+		}
+	}
+	if(meatZones contains loc)
+	{
+		goal = "meat";
+	}
+	else if(itemZones contains loc)
+	{
+		goal = "item";
+	}
+	else if(initZones contains loc)
+	{
+		goal = "init";
+	}
+	else if(mpZones contains loc)
+	{
+		goal = "mp";
+	}
+	else
+	{
+		//if for some reason we make it into the location getCitizenZone and it's not in any of the defined zones, get the item buff
```

Comment:
I think it would be better to abort than get the item buff, which will try to adventure elsewhere. I think this is unexpected, it is an error condition?